### ISSUE TITLE
Incorrect type for spreadConstraints value in Redis chart

### ISSUE THREAD

**Issue by mschuchard** [bitnami/charts#8304]
**Which chart**:
Redis 15.4.1 and current version

**Describe the bug**
The `spreadContraints` value is specified with a Map `{}` type, but it should be a List of Maps according to Kubernetes documentation. This is confirmed by the example in the Redis chart documentation:

```
  ## @param replica.spreadConstraints Spread Constraints for Redis&trade; replicas pod assignment
  ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
  ## E.g.
  ## spreadConstraints:
  ##   - maxSkew: 1
  ##     topologyKey: node
  ##     whenUnsatisfiable: DoNotSchedule
```

When using this as a value, the behavior is still mostly correct, but the expected warnings are generated during Golang coalescing:

> coalesce.go:200: warning: cannot overwrite table with non table for spreadConstraints (map[])

**To Reproduce**
Specify the `spreadContraints` example value in the documentation as an input.

**Expected behavior**
No warnings.

**Version of Helm**:
3.6.3

**Comment by javsalgar** [comment]
Hi,

Thanks for spotting the issue. Would you like to contribute with a PR? If not we will work on it.

**Comment by marcosbc** [comment]
Hi @mschuchard, what is happening there is that it is creating a dictionary like the following:

```
value:
- maxSkew: 1
  topologyKey: node
  whenUnsatisfiable: DoNotSchedule
context: $
```

And then passing that as input to the [`common.tplvalues.render` function](https://github.com/bitnami/charts/blob/master/bitnami/common/templates/_tplvalues.tpl), which would add it to the `topologySpreadConstraints` object for pods.

That said, it is my understanding that the following changes to the README would technically be enough:

```diff
diff --git a/bitnami/redis/values.yaml b/bitnami/redis/values.yaml
index 51328722f..b5ed83a4d 100644
--- a/bitnami/redis/values.yaml
+++ b/bitnami/redis/values.yaml
@@ -315,7 +315,7 @@ master:
   ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
   ##
   tolerations: []
-  ## @param master.spreadConstraints Spread Constraints for Redis&trade; master pod assignment
+  ## @param master.spreadConstraints [array] Spread Constraints for Redis&trade; master pod assignment
   ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
   ## E.g.
   ## spreadConstraints:
@@ -323,7 +323,7 @@ master:
   ##     topologyKey: node
   ##     whenUnsatisfiable: DoNotSchedule
   ##
-  spreadConstraints: {}
+  spreadConstraints: []
   ## @param master.lifecycleHooks for the Redis&trade; master container(s) to automate configuration before or after startup
   ##
   lifecycleHooks: {}
@@ -629,7 +629,7 @@ replica:
   ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
   ##
   tolerations: []
-  ## @param replica.spreadConstraints Spread Constraints for Redis&trade; replicas pod assignment
+  ## @param replica.spreadConstraints [array] Spread Constraints for Redis&trade; replicas pod assignment
   ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
   ## E.g.
   ## spreadConstraints:
@@ -637,7 +637,7 @@ replica:
   ##     topologyKey: node
   ##     whenUnsatisfiable: DoNotSchedule
   ##
-  spreadConstraints: {}
+  spreadConstraints: []
   ## @param replica.lifecycleHooks for the Redis&trade; replica container(s) to automate configuration before or after startup
   ##
   lifecycleHooks: {}
```

I applied those changes, and everything rendered fine.

If I then apply the sample configuration in both places, it continues to render fine:

```yaml
  spreadConstraints:
    - maxSkew: 1
      topologyKey: node
      whenUnsatisfiable: DoNotSchedule
```

So it seems like the only thing that needs to be fixed is the descriptions and default values in the `values.yaml`. Let me know if I'm missing anything.

**Comment by mschuchard** [comment]
Yeah I figured that was the output of the constructor, but did not understand the passing as an argument to that function, so thanks for the explanation.

I agree it renders fine, but I thought the go colasece warnings were related somehow. If they are not, then feel free to ignore that.

I also totally agree with the documentation update. My only question is whether the type is technically `array` (YAML) or `list`  (Go), but you would probably know before me.

**Comment by marcosbc** [comment]
I've sent a fix for the README in #8426. Once merged, this issue will automatically close.
